app/src/main/java/com/gmail/sanovikov71/githubclient/data/DataService.java

```diff
@@ -20,83 +24,136 @@
 import retrofit2.converter.gson.GsonConverterFactory;
 
 import java.util.List;
-import java.util.Objects;
 
 public class DataService extends Service {
 
     private static final String API_URL = "https://api.github.com/";
+    private static final String TAG = "DataService";
 
-    private RetrofitService mGithub;
+    private GithubApi mGithub;
 
     @Override
     public void onCreate() {
         Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl(API_URL)
                 .addConverterFactory(GsonConverterFactory.create())
                 .build();
-        mGithub = retrofit.create(RetrofitService.class);
+        mGithub = retrofit.create(GithubApi.class);
     }
 
-//    public static final String ACTION_FETCH = "com.gmail.sanovikov71.githubclient.ACTION_FETCH";
-
-//    @Override
-//    public int onStartCommand(Intent intent, int flags, int startId) {
-//        if (ACTION_FETCH.equals(intent.getAction())) {
-//            fetchUsers();
-//        }
-//
-//        return START_NOT_STICKY;
-//    }
-
-    public void fetchUsers(final UiElement ui) {
+    public void fetchUsers(final ProgressBarUiElement ui) {
         fetch(ui, 0);
     }
 
-    public void fetchMoreUsers(final UiElement ui, int offset) {
-        Log.i("Novikov", "fetchMoreUsers with offset: " + offset);
+    public void fetchMoreUsers(final ProgressBarUiElement ui, int offset) {
         fetch(ui, offset);
     }
 
-    public void fetch(final UiElement ui, int since) {
-        System.out.println("Novikov fetch");
-        Log.i("Novikov", "fetch");
+    public void fetch(final ProgressBarUiElement ui, int since) {
```

Comment:
It seems wrong that Service has a ref to a ProgressBarUiElement which is ui element. You may leak something.